src/main/resources/xslt/outputs/odt/models.xsl

```diff
@@ -742,35 +784,48 @@
 				<text:p text:style-name="Help"><xsl:value-of select="$instructionLabel"/></text:p>
 			</xsl:when>
 		</xsl:choose>		
-		
+				
+		<!-- Go to the Calculated Variable -->
 		<xsl:apply-templates select="eno:child-fields($source-context)" mode="source">
 			<xsl:with-param name="driver" select="." tunnel="yes"/>
-		</xsl:apply-templates>		
+		</xsl:apply-templates>
+				
 	</xsl:template>
 	
+
 	<xd:doc>
 		<xd:desc>
 			<xd:p>Template named:replaceVariablesInFormula.</xd:p>
-			<xd:p>It replaces variables in a formula.</xd:p>
+			<xd:p>It replaces variables in a all formula (control, instruction, filter).</xd:p>
 		</xd:desc>
 	</xd:doc>
 	<xsl:template name="replaceVariablesInFormula">
 		<xsl:param name="source-context" as="item()" tunnel="yes"/>
 		<xsl:param name="formula"/>
 		<xsl:param name="variables" as="xs:string*"/>
 		
-		<xsl:variable name="regex1" select="'number\(if\s\(//'" as="xs:string"/>
-		<xsl:variable name="regex2" select="'=''''\)\sthen\s''0''\selse\s//'" as="xs:string"/>
-		<xsl:variable name="regex3" select="'\)'" as="xs:string"/>
+		<xsl:variable name="regex1" select="$conditioning-variable-begin" as="xs:string"/>
+		<xsl:variable name="regex2" select="$conditioning-variable-end" as="xs:string"/>
+		<xsl:variable name="regex3" select="'number\(if\s\(//'" as="xs:string"/>
+		<xsl:variable name="regex4" select="'=''''\)\sthen\s''0''\selse\s//'" as="xs:string"/>
+		<xsl:variable name="regex5" select="'\)'" as="xs:string"/>
+		<xsl:variable name="regexA" select="concat($regex1,$variables[1],$regex2)"/>				
+		<xsl:variable name="regexB" select="concat($regex3,$variables[1],$regex4,$variables[1],$regex5)"/>
+		<xsl:variable name="regexC" select="concat('//',$variables[1])"/>
+		<xsl:variable name="expressionToReplace" select="concat('^',enoodt:get-business-name($source-context,$variables[1]))"/>

```

Comment:
Take a look at the xsl analyze-string instruction (you should find example in the v2beta branch, in the mw2xhtml.xsl stylesheet), it could simplify some regexp manipulation.